src/plugins/bfetch/common/bfetch_error.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { i18n } from '@kbn/i18n';
+
+/**
+ * Error thrown when xhr request fails
+ * @public
+ */
+export class BfetchRequestError extends Error {
+  /**
+   * constructor
+   * @param code - Xhr error code
+   */
+  constructor(code: number) {
+    const message =
+      code === 0
+        ? i18n.translate('bfetch.networkError', {
+            defaultMessage: 'Check your network connection and try again.',
+          })
+        : i18n.translate('bfetch.networkErrorWithStatus', {
```

Comment:
Just wanted to add that in testing I was able to get a 503 here when the Kibana server was starting up, and a 500 when ES was down. . . . Maybe we can add support for those at some point, because an arbitrary error code, while helpful for debugging, might not be helpful to an end user.

Reply:
Oh, also, I am not sure if it's possible, but it would be really nice if we could distinguish between "Kibana server is down" and "Network is unavailable", but in my testing I didn't see a difference in the error message.

Reply:
I think its possible to discern between `network down` and `kibana down` but it would require a bit more work - https://github.com/elastic/kibana/issues/146092

>Maybe we can add support for those at some point, because an arbitrary error code, while helpful for debugging, might not be helpful to an end user.

Are we certain that 500 always means ES is down and 503 always means kibana is starting up?